classes/MarkupConversionHelper.inc.php

```diff
@@ -401,6 +443,42 @@ public function addFileToSubmissionGalley($existing_galley_by_labels, $submissio
 		return $insertedFile;
 	}
 
+	/**
+	 * Extract HTML archive and attach media files to XML document (dependent files)
+	 *
+	 * @param $context Journal
+	 * @param $submission Submission
+	 * @param $submissionFile Submission File
+	 * @param $stage int stage
+	 * @param $filePath string Path to HTML archive
+	 * @param $params array Additional parameters (assoc type, assoc id)
+	 *
+	 * @return void
+	 */
+	protected function _processHtmlMediaFiles($context, $submission, $submissionFile, $stage, $htmlArchiveFilePath) {
+		// Extract the html zip archive
+		$message = '';
+		$destination = sys_get_temp_dir() . '/' . uniqid();
+		if (!$this->zipArchiveExtract($htmlArchiveFilePath, $destination, $message)) {
+			echo __(
```

Comment:
I see elsewhere you're using `fatalError` for error handling -- I think it would make sense to do the same here, no?

Reply:
Since this is a helper function I am reluctant to stop execution. That's the reason why I preferred returning false. Should I replace it by a call to `fatalError` ?

Reply:
I think a well-behaved helper function should either stop dead, or (better yet) provide error information to the caller. This is currently a mix of both.

Reply:
updated.